### ISSUE TITLE
feat(Material): Material You Support

### ISSUE THREAD

**Issue by intellix** [angular/components#22738]
#### Feature Description

Material You was announced a couple of days ago: https://www.youtube.com/watch?v=UHQPdP8qgrk which looks to be a visual redesign of the Material design system. 

It looks like it's going to become a little less opinionated in various aspects and that'd be great for those currently using Material today as a UI Library and having to hack the hell out of it to make it deviate slightly.

Quote from the description:
> We’re introducing a new visual redesign that gives you the power to customize the look and feel of your phone across the whole system. With smoother animations, more intuitive notifications, quick settings and new widgets for your favorite people, Android will feel more personal and alive than ever before.

My guess is that this'll arrive upstream via MDC which is already being implemented. Will this be considered?

**Comment by crisbeto** [comment]
We should get this automatically once we've finished the transition to MDC components.

**Comment by CiriousJoker** [comment]
Will this be "opt in" in any way? Not that I dislike it, but for my web app it's perhaps not as fitting and I'd rather not stay on an old version just to keep the design

**Comment by HaidarZ** [comment]
Any progress on this issue or is there any roadmap for this upgrade ?

**Comment by aastrouski** [comment]
Any updates here?

**Comment by keatkeat87** [comment]
![image](https://user-images.githubusercontent.com/8691013/168807254-a09d7dcc-bce4-47cf-aa06-8d402b7dc35d.png)
[from issue: When is Material Web 3 Coming?](https://github.com/material-components/material-web/issues/3050)

I am really confused about that. can someone give a clear answer here?
1. Angular Material mdc is depend on [material-components-web](https://github.com/material-components/material-components-web)?
2. Material Team is not focused on material-components-web anymore, they turn to [material-web](https://github.com/material-components/material-web)?
3. Will Angular Material turn to material-web also or not going to have a Material You?

**Comment by jelbourn** [comment]
We're working directly with the Material Design group inside Google towards supporting design updates. No ETA just at this moment, but it's definitely something we're working towards.

**Comment by hillin** [comment]
Glad to see I'm not the only one who's confused. I've been watching and waiting for a design update, literally for years - mainly for the density feature, but M3 is surely a welcome addition.

From what I've read:

- The Angular material team has been working on integrating with the MDC components for **over three years** and still actively working on it, known as the *experiment* components. They were [planning to stablize and release them in version 14](https://github.com/angular/components/issues/23923) but since v14 is imminent I doubt it's still on track;
- MDC is based on Material Design 2 (M2) and is not gonna be updated for Material You (M3) (https://github.com/material-components/material-web/issues/3050#issuecomment-1011464627)
- A new set of components, known as the MWC, is based on M3, which will launch later this year;
- It doesn't seem Angular material has started to integrate with MWC yet
- as @crisbeto stated, [we'll get M3 when the transition to MDC is complete](https://github.com/angular/components/issues/22738#issuecomment-846147435), seemingly contradicts with the facts above
- It seems Angular material has not started or is still in the early stage of integrating with MWC, so can I assume we are not getting M3 in the near future?

**Comment by MikaStark** [comment]
Angular Material MDC migration has been a running gag in my company for years. Some kind of "When are you planning to release this feature ? -- Oh you know, when Angular Material will use MDC".
It's a gentle joke but actually it's a sad situation because I feel (and I guess not to be the only one) like Angular Material will never propose real new features anymore (like two-direction drag-n-drop, more theming customization, more card capabilities or even time picker) and every real improvment are only reserved for Android and, more generally, for mobile devs. I'm aware that it's an open source project and the team has limited resource (just like every company) but it's hard to not feel impatience and frustration. That said, I will keep supporting and using Angular/Material because I love it. But guys please, do not let us down.

**Comment by CnxYankoNikolov** [comment]
Is there any new information on the confusion in this thread? There doesn't seem to be any indication that Angular Material will support Material You (Material Design 3) from the fact that it references the old MDC implementation. We are looking into building a design on top of Material Design 3 and Angular Material, so I was wondering if this would be possible in the near future or if we should scrap that idea for the next year or 2?

**Comment by siva636** [comment]
The Material components offered now with Angular is the legacy components, which was released before Material 2 specification was out.

According to unofficial rumours the Material 2 components might be made available with Angular 15 release latter this year, but still it is not confirmed.

Having Material 3 components with Angular may not be a reality for the next several years. It might not happen at all.

**Comment by michaelfaith** [comment]
> According to unofficial rumours the Material 2 components might be made available with Angular 15 release latter this year, but still it is not confirmed.
Looking at v15 "next" release notes would suggest it's very likely they'll be rolled out with the release.  Now, that's not to say it won't go through a "Developer Preview" period like some of the big new features that've been released recently, but even in that case, it seems very close to seeing the light of day: https://github.com/angular/components/releases/tag/15.0.0-next.0

**Comment by jelbourn** [comment]
v15 will promote the MDC-based components ot stable. These components are a stepping stone to future iterations of the design system (M3). If you peek on ongoing PRs, you'll see the theming mixins for many of the components transitioning to design tokens, which is the basis for M3.

**Comment by michaelfaith** [comment]
I noticed that!  I'm excited to see a greater use of CSS Custom props.  Do you foresee that impacting how teams configure the overall theme using the Sass theming API?

**Comment by jelbourn** [comment]
Yeah, any support for M3 will be part of a larger update to Angular Material's theming system. While the design isn't fully complete, it will likely involve a lot more customization options.

**Comment by zarghamkhandev** [comment]
@jelbourn  is there any way that one could contribute to moving forward with this change?

**Comment by jelbourn** [comment]
Unfortunately no, we're still waiting on deliverables from the Material Design folks to be able to move forward on the Angular side.

**Comment by alex737876** [comment]
@jelbourn Thanks, Are there already specific issue(s) on the Material Design side that we can point to here? 
Maybe the community wants to support them to speed things up here :)

**Comment by jelbourn** [comment]
I don't think so, since the changes relate to the underlying Material Design token definitions for web which I don't think have been formally published anywhere.

**Comment by floinay** [comment]
Any updates?

**Comment by fromsomnia** [comment]
Some [incremental work](https://github.com/angular/components/pull/26674/files) transitioning to design tokens is associated with the Angular 16 release.

I also spotted [the barest of hopeful hints](https://twitter.com/Bitcollage/status/1632441595084832768) from @mgechev of the Angular team on the general Material You development arc (along with further context that may be of interest).

Additionally, in the broader context of Angular 16, [a brief Angular 16 release summary I found insightful](https://betterprogramming.pub/angular-16-whats-new-and-what-to-expect-e14e15e500f8) also notes this Material You development.

**Comment by isaiaseg** [comment]
Are we getting Material 3 styles in Angular Material?

**Comment by ansarizafar** [comment]
Is there any update on Angular Material 3 support?

**Comment by shivarajnaidu** [comment]
Any update about Material 3 ?

**Comment by fromsomnia** [comment]
In a [recent tweet](https://twitter.com/Bitcollage/status/1685988680438054912) @SerkanSipahi mentioned a [PR](https://github.com/angular/components/pull/27532) for the initial implementation of Material 3 theme objects that has since been merged as part of a current [Angular Material benchmark](https://angular.io/guide/roadmap#token-based-theming-apis) in the Angular Roadmap towards enabling Material 3 capabilities. As another marker of progress also note that legacy Angular Material components (pre-MDC) [will be dropped in Angular 17](https://twitter.com/yjaaidi/status/1653374242581274626).

**Comment by justAsascha** [comment]
Any updates?

**Comment by naveedahmed1** [comment]
https://twitter.com/materialdesign/status/1714378297696461160?t=hDBDncKLA-qWuXPBBL6PCA&s=19

**Comment by ciriousjoker** [comment]
Can someone please explain the relation between Angular Material and Material Web? Is Angular Material going to transition from Material Components Web -> Material Web next or will those be kept completely separate? What's the plan here?

**Comment by SerkanSipahi** [comment]
cc @ciriousjoker 

Material Web uses the Lit Framework as the core engine and Angular Material uses the Angular Framework as the core engine.

**Comment by ciriousjoker** [comment]
@SerkanSipahi You can use web components (which lit produces afaik) in Angular as well though. I'm assuming the benefit is therefore mostly in the convenience of using them and probably a better integration?

Given the speed at which Material Web seems(!!) to have been implemented relative to MDC, is that project a higher priority at Google?

I'm trying to figure out if it makes sense to use Material Web in Angular even though the readme currently suggests using Angular Material. I'm essentially speculating that Google might pull the plug on Angular Material and/or the MDC it's built on in the future and the overhead of using Material Web in Angular might therefore be worthwhile.

**Comment by ciriousjoker** [comment]
> Feels like Angular is the next on https://killedbygoogle.com/

Felt like it a while ago when react had that huge jump with SSR and hooks and all the other shiny things popped up (shelter, solid, ...), but v15+ seems like Google actually puts in resources again, so I wouldn't worry about that.

off topic though...

**Comment by fromsomnia** [comment]
During the [Angular Special Event](https://www.youtube.com/watch?v=Wq6GpTZ7AX0) on Nov 6, 2023, @mmalerba of the Angular team [mentioned](https://www.youtube.com/clip/UgkxPxhTWen956nsGLJcdMkEkY00snBbTYQ6) that we might see Material 3 as soon as a minor version of v17. Later during the same event @simonaco of the Angular team also [confirmed](https://youtube.com/clip/UgkxX06xg0gyDAIl9ujoX0fvFoCF-v45obJg?si=Rw5uAe-OIeZ_D0ai) that Material 3 is one of the Angular team's priorities moving beyond [v17](https://blog.angular.io/tagged/angular-v17) (Nov 8) and the advent of [angular.dev](https://angular.dev/).

**Comment by Luiza-CR** [comment]
Any updates?

**Comment by hillin** [comment]
Apparently recent [releases](https://github.com/angular/components/releases/) have shown some great progress towards M3. It's just I didn't find an instruction of how to use or opt-in these new features.

**Comment by danielalexis** [comment]
> Apparently recent [releases](https://github.com/angular/components/releases/) have shown some great progress towards M3. It's just I didn't find an instruction of how to use or opt-in these new features.

Run `npm i @angular/material@17.1.0-next.2` or `npm i @angular/material@next` in your project. This will install the version with the M3 components, should be noted that this is obviously not a stable version

**Comment by ChrTall** [comment]
Are the new design tokens replacing the sass api or are there going to be breaking changes in the sass api?
I have some components that are styled based on the provided material theme and use partial scss files.

**Comment by morfees** [comment]
> Run `npm i @angular/material@17.1.0-next.2` or `npm i @angular/material@next` in your project. This will install the version with the M3 components, should be noted that this is obviously not a stable version

Thanks for the tip. Do you know if there is an entry point for the M3 components ? because I just installed the latest pre-release version (angular/material@17.1.0-next.3) and it still has the old design by default.

**Comment by danielalexis** [comment]
Instructions to use Material 3 are listed in material-experimental:
https://github.com/angular/components/tree/main/src/material-experimental

**Comment by aastrouski** [comment]
> Instructions to use Material 3 are listed in material-experimental: https://github.com/angular/components/tree/main/src/material-experimental

These are instructions for Material 2 MDC components, which have been released with Angular 15.

**Comment by dbesiryan** [comment]
Hey folks, 
any instruction/update on how to use the M3 components in Angular (RC)??

**Comment by miguelmoya1** [comment]
Hi everyone,

I'm writing to follow up on the release of angular/components version 17.1.0 yesterday. While this version introduces support for M3, it seems to default to M2.

Could you shed some light on how to enable and utilize M3 with this new release? Any guidance or documentation would be greatly appreciated.

Thanks in advance,

**Comment by shimonsalamon** [comment]
As per this [README.md](https://github.com/angular/components/blob/main/src/material-experimental/README.md) we need to add the theme and typography mixins to the Sass.

**I wasn't able to get it to work.**

```scss
  @use '@angular/material' as mat;
  @use '@angular/material-experimental' as mat-experimental;

  $my-primary: mat.define-palette(mat.$indigo-palette);
  $my-accent: mat.define-palette(mat.$pink-palette, A200, A100, A400);
  $my-theme: mat.define-light-theme((
    color: (
      primary: $my-primary,
      accent: $my-accent
    ),
    // Using `define-mdc-typography-config` rather than `define-typography-config` generates a
    // typography config directly from the official Material Design styles. This includes using
    // `rem`-based measurements rather than `px`-based ones as the spec recommends.
    typography: mat-experimental.define-mdc-typography-config(),
    // The density level to use in this theme, defaults to 0 if not specified.
    density: 0
  ));

  @include mat-experimental.all-mdc-component-themes($my-theme);
```

**Comment by aastrouski** [comment]
There are no M3 components in 17.1.

**Comment by eneajaho** [comment]
> As per this [README.md](https://github.com/angular/components/blob/main/src/material-experimental/README.md) we need to add the theme and typography mixins to the Sass.
> 
> **I wasn't able to get it to work.**
> 
> ```css-scss
>   @use '@angular/material' as mat;
>   @use '@angular/material-experimental' as mat-experimental;
> 
>   $my-primary: mat.define-palette(mat.$indigo-palette);
>   $my-accent: mat.define-palette(mat.$pink-palette, A200, A100, A400);
>   $my-theme: mat.define-light-theme((
>     color: (
>       primary: $my-primary,
>       accent: $my-accent
>     ),
>     // Using `define-mdc-typography-config` rather than `define-typography-config` generates a
>     // typography config directly from the official Material Design styles. This includes using
>     // `rem`-based measurements rather than `px`-based ones as the spec recommends.
>     typography: mat-experimental.define-mdc-typography-config(),
>     // The density level to use in this theme, defaults to 0 if not specified.
>     density: 0
>   ));
> 
>   @include mat-experimental.all-mdc-component-themes($my-theme);
> ```

I opened an issue for it: https://github.com/angular/components/issues/28465


There has been a lot of work going on for M3 

Check out the demo app: click on Use M3 theme and scroll on components to see what they look like. 
https://ng-comp-devapp.web.app/button

**Comment by rsi21** [comment]
Hi all,

Just for info, as pointed by @eneajaho , readme is not ready but looking at [theme-m3.scss](https://github.com/angular/components/blob/main/src/dev-app/theme-m3.scss) it is posible to test M3.

I just wanted to ask, as question comes often, is there a way to follow the advancement and tasks remaining around M3 (components migrated, in progress or not started)? Something like in material web project for example:

![image](https://github.com/angular/components/assets/14235105/0808ea16-689a-422a-8863-4bcb45f16ac7)

It could help for visibility and also to have may be feedbacks around components fully migrated.

It is just a question :)

**Comment by eneajaho** [comment]
> Hi all,
> 
> Just for info, as pointed by @eneajaho , readme is not ready but looking at [theme-m3.scss](https://github.com/angular/components/blob/main/src/dev-app/theme-m3.scss) it is posible to test M3.
> 
> I just wanted to ask, as question comes often, is there a way to follow the advancement and tasks remaining around M3 (components migrated, in progress or not started)? Something like in material web project for example:
> 
> ![image](https://private-user-images.githubusercontent.com/14235105/299286853-0808ea16-689a-422a-8863-4bcb45f16ac7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYwOTQyNDcsIm5iZiI6MTcwNjA5Mzk0NywicGF0aCI6Ii8xNDIzNTEwNS8yOTkyODY4NTMtMDgwOGVhMTYtNjg5YS00MjJhLTg4NjMtNGJjYjQ1ZjE2YWM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI0VDEwNTkwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM2ZThjNWViNmRmYzliMDQ4MWQxZTEyYjJiMDM4YzhjNTA3NmFiNmI3ZGFkMzljOGQxMWIwNWE1MmY1NWZlZDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Nk8k-SGbzJwyY_yjW2VQA-BcYih4o32-7IEcbeum0YU)
> 
> It could help for visibility and also to have may be feedbacks around components fully migrated.
> 
> It is just a question :)

Hi
I managed to get it working

- Repo: https://github.com/eneajaho/material-3-demo
- Commit https://github.com/eneajaho/material-3-demo/commit/3f8e742967ade4b0c685c07b4ce727b1ba82a08e

https://github.com/angular/components/assets/25394362/36962f4a-ba6c-4523-94cb-98fce405fd1a

**Comment by rsi21** [comment]
Hi,

Since version 17.2.0, we have Material 3 and all the guides to migrate and that's great :) (But in experimental of course.)

But i was just wondering, if as said in the M3 migration guide " The team plans to stabilize support for M3 after a brief period in experimental in order to get feedback on the design and API."

Do we know if it is still plan for a final delivery in next version 17.3? Otherwise i imagine it won't be available until Angular 18 ?
(As i assume 17.3 will  be the latest in Angular 17)

Thanks

**Comment by hillin** [comment]
Is there a project for tracking the progress of m3 implementation?
For instance I found the slider component is still in m2 style, as of version 17.3.5. I would like to know which components are m3-ready.

I found an [old project for MDC](https://github.com/angular/components/projects/25) but apparently it's not for m3.

**Comment by amysorto** [comment]
Closing this issue since [M3 is available in experimental as of 17.2](https://blog.angular.io/material-3-experimental-support-in-angular-17-2-8e681dde650e) and will be stable soon. Here is the [guide for migrating to M3](https://material.angular.io/guide/material-3).

@hillin this is the [M3 project board](https://github.com/orgs/angular/projects/35) we are using

**Comment by hillin** [comment]
Thanks @amysorto, but the project board link is broken (404). Do I need certain priviledge to see it?
